### DISCUSSION TITLE
Cannot compile oin openSUSE 15.4

### DISCUSSION THREAD

**Discussion by xavierbaez** [Audio4Linux/JDSP4Linux#116] [General]
Hello. Can someone please help me install this on openSUSE?

```
g++: error: unrecognized command line option ‘-std=gnu++2a’; did you mean ‘-std=gnu++03’?
make[1]: *** [Makefile:2638: scheduler.o] Error 1
make[1]: *** Waiting for unfinished jobs....
g++: error: unrecognized command line option ‘-std=gnu++2a’; did you mean ‘-std=gnu++03’?
g++: error: unrecognized command line option ‘-std=gnu++2a’; did you mean ‘-std=gnu++03’?
make[1]: *** [Makefile:2661: threadpool_scheduler.o] Error 1
make[1]: *** [Makefile:2664: qcustomplot.o] Error 1
g++: error: unrecognized command line option ‘-std=gnu++2a’; did you mean ‘-std=gnu++03’?
make[1]: *** [Makefile:2671: writer.o] Error 1
make[1]: Leaving directory '/home/xavier/Programs/aur/JDSP4Linux/build/src'
make: *** [Makefile:73: sub-src-make_first] Error 2
```

**Comment by timschneeb** [comment]
`-std=gnu++2a` is supposed to enable C++20 support in the compiler. If it complains about that option not existing, that means that the g++ compiler installed on your computer is too old.

You need to download and install at least version 10.0 of the gcc/g++ compiler to compile this app.


__________________
Also, after installing another compiler version, you might end up with multiple compilers on your machine. In that case, you need to specify which compiler version you want to use. You can do this via environment variables in the terminal when building this app.

Example for gcc version 11:
```
export CC=/usr/bin/gcc-11
export CXX=/usr/bin/g++-11
qmake .
make
```
